staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/types_jsonschema.go

```diff
@@ -168,7 +168,7 @@ type JSONSchemaProps struct {
 }
 
 // ValidationRules describe a list of validation rules for expression validation
-//  +listType=atomic
+//  +listMapKey=rule
```

Comment:
a key for atomic does not make sense. I don't think we have a map here and not a set. Atomic sounds fine.

Reply:
https://github.com/kubernetes/kubernetes/pull/105477#discussion_r740558116
From Joe's comment:
```
I'm a bit torn on this one. On one hand, I don't really see the schema of a CRD as being something that can be sub-divided across multiple owners working independently. On the other hand, we made CEL validation rules a list of rules to allow each rule to be authored independently.

Since it looks like we already allow for non-atomic modification of OpenAPIv3 schemas, I'm slightly in favor of listMapKey=rule, but would like to hear what others think.
```

Reply:
Updated to use 
```
// +listType=map
// +listMapKey=rule
```
for now. Please let me know if atomic is strongly preferred.